lib/compat/wordpress-6.8/template-parts.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * Adds the default template part areas to the REST API index.
+ *
+ * This function exposes the default template part areas through the WordPress REST API.
+ * Note: This function backports into the wp-includes/rest-api/class-wp-rest-server.php file.
+ *
+ * @param WP_REST_Response $response REST API response.
+ * @return WP_REST_Response Modified REST API response with default template part areas.
+ */
+function gutenberg_add_default_template_part_areas_to_index( WP_REST_Response $response ) {
+	$response->data['default_template_part_areas'] = get_allowed_block_template_part_areas();
+	return $response;
+}
+
+add_action( 'rest_index', 'gutenberg_add_default_template_part_areas_to_index' );
```

Comment:
Nit: the [`rest_index` is a filter](https://developer.wordpress.org/reference/hooks/rest_index/), let's use `add_filter` here.

We could also combine two callbacks into one, but that's optional.

Reply:
Fixed with ffdcc1d585c6879dbd07c817971480df43be65f3